examples/authorized_api/authorization.go

```diff
@@ -3,10 +3,58 @@ package main
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
 	"io/ioutil"
 	"net/http"
 )
 
+func ZebedeeMustPermitUser(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		status, err := zebedeeValidateUser()
+		if err != nil {
+			if status == "401 Unauthorized" {
+				http.Error(w, err.Error(), http.StatusUnauthorized)
+			} else {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+			}
+			return
+		} else {
+			w.WriteHeader(200)
+			message := "[\"DELETE\", \"READ\", \"CREATE\", \"UPDATE\"]"
+			fmt.Fprintf(w, message)
+		}
+		handler(w, r)
+	}
+}
+
+func zebedeeValidateUser() (string, error) {
+	type Permissions struct {
+		Permissions string `bson:"message" json:"message"`
+	}
+	var permissions Permissions
+	config := NewConfig()
+	response, err := http.Get(config.authorizationServiceUrl + "/userInstancePermissions")
+	status := response.Status
+	if err != nil {
+		return status, err
+	}
+	body, err := ioutil.ReadAll(response.Body)
```

Comment:
You need a `defer response.Body.close()` here.

Reply:
done, thanks